### DISCUSSION TITLE
Add 1-column view (sidebar view without the sidebar)

### DISCUSSION THREAD

**Discussion by Kanecaine** [home-assistant/frontend#18758] [Dashboard]
As for now we have 3 view options:
- Masonry (default)
- Panel
- Sidebar

The Masonry layout is great for automatic dashboards but (at least for me) it is not really usable when i take over control by creating my own dashboards. There i have to get along with panel (1 card) and sidebar view. The panel layout isn't just limited to only 1 card because this card could be e.g. a vertical-stack-card which can contain other cards. I often end with a mix of vertical and horizontal stacked cards. 
The problem with this is that all cards are underneath this topmost layout card. A complex view with many cards is difficult to edit: first you have to scroll all the way down to reach the one and only edit button and then you have to go through all the nested cards till you reach the one you want to edit.

Using the sidebar view doesn't come with this downside: here you have literally an 1-column layout at the main area. You can place the cards without the need of nesting them all: e.g. multiple horizontal-stack-cards do not need a superior vertical-stack-card. But of course you have the sidebar.

I would like to have the "best of both worlds": a view that looks like a panel view but behaves like the main area of the sidebar view when editing it. Literally an 1-column view. When someone comes to this maybe it makes sense to also add an 2-column and 3-columns view.

**Comment by Kanecaine** [comment]
By chance i did not move any card to the sidebar in a sidebar view for the first time ... as it turns out in that case the sidebar isn't shown at all. Really? Is that true? Then my feature request is absolutely obsolete, sorry for bothering.

**Comment by ildar170975** [comment]
>  it turns out in that case the sidebar isn't shown at all.

Not exactly - it is just empty.
Do not think it is what you need.

![изображение](https://github.com/home-assistant/frontend/assets/71872483/adb95573-d713-4460-871f-0e51cb79299f)

**Reply by Kanecaine** [reply]
Thanks for your reply - that was exactly what i assumed. Therefore i did set the width of the sidebar to 0px with a card mod theme. This solved my issue in a hacky way. BUT when i remove the css style than it still shows no sidebar:

<img width="1206" alt="Bildschirmfoto 2023-11-25 um 22 37 40" src="https://github.com/home-assistant/frontend/assets/99053167/0eaf77e8-301f-4ae0-b120-bc7b15d55818">

I reloaded themes, cleared all caches and proved it on another device: when there is no card the sidebar isn't shown. In editor mode i still can assign a card. After removing it again the main area uses the whole space - no empty sidebar. Exactly what i want.

edit: here the screen with dom and styles:

<img width="1391" alt="Bildschirmfoto 2023-11-25 um 22 52 30" src="https://github.com/home-assistant/frontend/assets/99053167/5c1c1d18-597c-47e3-b441-0d29ecc2dc12">

**Comment by Kanecaine** [comment]
Unfortunately the observed behavior is not consistently: sometimes an empty sidebar is visible and sometimes not. I'm not shure how to reproduce it since it seems to depend on browser and / or view content. Same view opened in the companion app on android shows the sidebar in contrast to Firefox on Mac where it not visible. Same dashboard shows in one view an empty sidebar and in another view no sidebar on the same device and browser.

**So my request ist up-to-date again but has to be substantiated:**

- Please provide the ability for a 1- (2-, 3-)column view as described in the first post or
- Please fix / adapt the behavior of the sidebar view so that the sidebar is (consistently) not shown if there's no card assigned to it
